### PR TITLE
BAEL 3986 - mapping lists using ModelMapper

### PR DESCRIPTION
**Pull request by sasam0320**

### CODE REVIEW

File: model-mapper/src/Main.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+
+/**
+ * @author sasam0320
+ * @date 4/18/2020
+ */
+
+public class Main {
+
+    public static void main(String[] args) {
```
Comment (rpvilao) [inline model-mapper/src/Main.java:19]: we don't use printlns or main methods. usually, we create unit tests or just methods in general when necessary, maybe we can do the same here.

File: model-mapper/src/Main.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import com.baeldung.model.User;
```
Comment (rpvilao) [inline model-mapper/src/Main.java:1]: we don't create new modules unless necessary and most certainly not in the root, so let's move the code to an existing module. I think a good candidate would be java-collections-conversions-2. do not create a new maven module, but rather use the existing one and create a package called modelmapper for example

File: model-mapper/src/com/baeldung/util/UserPropertyMap.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.modelmapper.Converter;
+import org.modelmapper.PropertyMap;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author sasam0320
+ * @date 4/18/2020
+ */
+
```
Comment (rpvilao) [inline model-mapper/src/com/baeldung/util/UserPropertyMap.java:16]: remove space.

File: model-mapper/src/com/baeldung/util/UserPropertyMap.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import com.baeldung.model.UserDTO;
+import com.baeldung.model.UserList;
+import org.modelmapper.AbstractConverter;
+import org.modelmapper.Converter;
+import org.modelmapper.PropertyMap;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author sasam0320
```
Comment (rpvilao) [inline model-mapper/src/com/baeldung/util/UserPropertyMap.java:13]: describe and use put your name and remove the date (I am not even sure that's a valid Javadoc tag)

File: model-mapper/src/com/baeldung/util/MapperUtil.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.baeldung.util;
+
+import org.modelmapper.ModelMapper;
+import org.modelmapper.convention.MatchingStrategies;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author sasam0320
```
Comment (rpvilao) [inline model-mapper/src/com/baeldung/util/MapperUtil.java:10]: same

File: model-mapper/src/com/baeldung/model/UserList.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package com.baeldung.model;
+
+import java.util.Collection;
+
+/**
+ * @author sasam0320
+ * @date 4/18/2020
+ */
+
```
Comment (rpvilao) [inline model-mapper/src/com/baeldung/model/UserList.java:9]: remove space

File: model-mapper/src/com/baeldung/model/UserDTO.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package com.baeldung.model;
+
+import java.util.List;
+
+/**
+ * @author sasam0320
```
Comment (rpvilao) [inline model-mapper/src/com/baeldung/model/UserDTO.java:6]: same

File: model-mapper/src/Main.java (hunk 2 above)
Comment (sasam0320) [reply model-mapper/src/Main.java:1]: does this mean that the code should be put in the test package of the module?

Comment (rpvilao) [reply model-mapper/src/Main.java:1]: no, we use both. code that matters on the src and test related stuff (test classes and possible helpers on the test)

Comment (sasam0320) [reply model-mapper/src/Main.java:1]: Ok. i almost forget. the model-mapper.jar dependency should be added to pom?

Comment (rpvilao) [reply model-mapper/src/Main.java:1]: we use maven:
https://mvnrepository.com/artifact/org.modelmapper/modelmapper

Comment (rpvilao) [reply model-mapper/src/Main.java:1]: I mean we need to because if not it won't build

Comment (sasam0320) [reply model-mapper/src/Main.java]: Code changes are made. Can you take a look?

File: java-collections-conversions-2/pom.xml [hunk 8]
```diff
@@ -20,6 +20,23 @@
             <artifactId>commons-lang3</artifactId>
             <version>${commons-lang3.version}</version>
         </dependency>
+		<dependency>
```
Comment (rpvilao) [inline java-collections-conversions-2/pom.xml:23]: fix indentation

File: java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/MapperUtil.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
+package com.baeldung.modelmapper;
+
+import org.modelmapper.ModelMapper;
+import org.modelmapper.convention.MatchingStrategies;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author sasam0320
```
Comment (rpvilao) [inline java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/MapperUtil.java:10]: fix

File: java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/MapperUtil.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+ * @description
```
Comment (rpvilao) [inline java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/MapperUtil.java:11]: I think you should take a look about Javadoc, you have tags that don't even exist

File: java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/User.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+    private String userId;
+    private String userName;
+    private String email;
+    private String contactNumber;
+    private String userType;
+
+    // Standard constructors, getters and setters
+
+    public User(){}
```
Comment (rpvilao) [inline java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/User.java:17]: indent

File: java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/UserDTO.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
+package com.baeldung.modelmapper;
+
+/**
+ * @author sasam0320
```
Comment (rpvilao) [inline java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/UserDTO.java:4]: fix javadoc

File: java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/UserListDTO.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
+package com.baeldung.modelmapper;
+
+import java.util.List;
+
+/**
+ * @author sasam0320
```
Comment (rpvilao) [inline java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/UserListDTO.java:6]: fix

File: java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/UserPropertyMap.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.modelmapper.AbstractConverter;
+import org.modelmapper.Converter;
+import org.modelmapper.PropertyMap;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author sasam0320
+ * @description
```
Comment (rpvilao) [inline java-collections-conversions-2/src/main/java/com/baeldung/modelmapper/UserPropertyMap.java:11]: fix

File: java-collections-conversions-2/src/test/java/com/baeldung/modelmapper/UserMappingTest.java [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * @sasam0320
+ * @description
```
Comment (rpvilao) [inline java-collections-conversions-2/src/test/java/com/baeldung/modelmapper/UserMappingTest.java:23]: fix

File: java-collections-conversions-2/src/test/java/com/baeldung/modelmapper/UserMappingTest.java [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+        mapper.addMappings(new UserPropertyMap());
+        users = new ArrayList();
+        users.add(new User("b100", "user1", "user1@baeldung.com", "111-222", "USER"));
+        users.add(new User("b101", "user2", "user2@baeldung.com", "111-333", "USER"));
+        users.add(new User("b102", "user3", "user3@baeldung.com", "111-444", "ADMIN"));
+
+    }
+
+    @Test
+    public void testMapIntegerList() {
```
Comment (rpvilao) [inline java-collections-conversions-2/src/test/java/com/baeldung/modelmapper/UserMappingTest.java:45]: user the baeldung conventions for test names. you can find the documentation in the drive

File: parent-java/pom.xml [hunk 17]
```diff
@@ -45,6 +45,9 @@
         <guava.version>23.0</guava.version>
         <commons.io.version>2.6</commons.io.version>
         <jmh.version>1.19</jmh.version>
+		<modelmapper.version>2.3.6</modelmapper.version>
```
Comment (rpvilao) [inline parent-java/pom.xml:48]: indent

File: parent-java/pom.xml [hunk 18]
```diff
@@ -45,6 +45,9 @@
         <guava.version>23.0</guava.version>
         <commons.io.version>2.6</commons.io.version>
         <jmh.version>1.19</jmh.version>
+        <modelmapper.version>2.3.6</modelmapper.version>
```
Comment (rpvilao) [inline parent-java/pom.xml:48]: update to 2.3.7

File: java-collections-conversions-2/src/test/java/com/baeldung/modelmapper/UsersListMappingUnitTest.java [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * This class has test methods of mapping Integer to Character list,
+ * mapping user list to DTO list using MapperUtil generic methods and Converter
+ *
+ * @author sasam0320
```
Comment (rpvilao) [inline java-collections-conversions-2/src/test/java/com/baeldung/modelmapper/UsersListMappingUnitTest.java:24]: let's use your name everywhere

**Pull request merged**